presto-main/src/main/java/com/facebook/presto/execution/DeduplicatingWarningSink.java

```diff
@@ -15,25 +15,26 @@
 
 import com.facebook.presto.spi.WarningCode;
 import com.google.common.collect.ImmutableList;
-import io.airlift.log.Logger;
 
-import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
 
 import static java.util.Objects.requireNonNull;
 
-public class LocalLoggingWarningSink
+public class DeduplicatingWarningSink
         implements WarningSink
 {
-    private static final Logger log = Logger.get(WarningSink.class);
+    private final Set<PrestoWarning> warnings = new LinkedHashSet<>();
```

Comment:
why not just `HashSet`?

Reply:
Not sure, I need to verify whether it is important to preserve the order of unique warnings, will find out and update.